api/app/src/test/kotlin/packit/integration/controllers/PacketControllerTest.kt

```diff
@@ -512,4 +530,86 @@ class PacketControllerTest : IntegrationTest()
         )
         assertBadRequest(result)
     }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["packet.manage"])
+    fun `user with packet manage can update read permission on roles`()
+    {
+        val packet = packetRepository.findAll().first()
+        val roleNamesToAdd = setOf("testRole1", "testRole2")
+        val roleNamesToRemove = setOf("testRole3", "testRole4")
+        roleRepository.saveAll(
+            roleNamesToAdd.map {
+                Role(name = it)
+            }
+        )
+        val rolesToRemove = roleRepository.saveAll(
+            roleNamesToRemove.map {
+                Role(name = it)
+            }
+        ).onEach {
+            it.rolePermissions = mutableListOf(
+                RolePermission(
+                    role = it,
+                    permission = permissionsRepository.findByName("packet.read")!!,
+                    packet = packet
+                )
+            )
+        }
+        roleRepository.saveAll(rolesToRemove)
+        val updatePacketReadRoles = jacksonObjectMapper().writeValueAsString(
+            UpdateReadRoles(
+                roleNamesToAdd = roleNamesToAdd,
+                roleNamesToRemove = roleNamesToRemove
+            )
+        )
+
+        val result = restTemplate.exchange(
+            "/packets/${packet.id}/read-permission",
+            HttpMethod.PUT,
+            getTokenizedHttpEntity(data = updatePacketReadRoles),
+            String::class.java
+        )
+
+        assertEquals(result.statusCode, HttpStatus.NO_CONTENT)
+        val addedRoles = roleRepository.findByNameIn(roleNamesToAdd.toList())
+        val removedRoles = roleRepository.findByNameIn(roleNamesToRemove.toList())
+        removedRoles.forEach {
+            assertEquals(
+                it.rolePermissions.size,
+                0
+            )
+        }
+        addedRoles.forEach {
+            assertEquals(
+                it.rolePermissions.first().packet?.id,
+                packet.id
+            )
+        }
+    }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["packet.manage"])
+    fun `can get roles and users for update for packet`()
+    {
+        val packet = packetRepository.findAll().first()
+        val result = restTemplate.exchange(
+            "/packets/${packet.id}/read-permission",
+            HttpMethod.GET,
+            getTokenizedHttpEntity(),
+            String::class.java
+        )
+
+        assertSuccess(result)
+
+        val body = jacksonObjectMapper().readValue(
+            result.body,
+            object : TypeReference<RolesAndUsersForReadUpdate>()
+            {}
+        )
+
+        assert(body.canRead is BasicRolesAndUsersDto)
+        assert(body.withRead is BasicRolesAndUsersDto)
+        assert(body.cantRead is BasicRolesAndUsersDto)
```

Comment:
These lines have no added value. All they do is check something that the type of `RolesAndUsersForReadUpdate` already tells you. The test above doing the PUT and GET seems enough.